### PR TITLE
Decompile jpegdecoder.c, padsetup.c and code_800C3C20.c

### PR DESCRIPTION
**Pull request by Random06457**
`jpegdecoder.c`: 1 non-matching left
`padsetup.c`: OK
`code_800C3C20.c`: OK

### CODE REVIEW

File: src/code/padsetup.c [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        osContGetQuery(status);
+
+        *outMask = 0;
+        for (i = 0; i < 4; i++) {
+            if (status[i].errno == 0) {
+                if (status[i].type == CONT_TYPE_NORMAL) {
+                    *outMask |= 1 << i;
+                }
+            } else {
+            }
```
Comment (Roman971) [inline src/code/padsetup.c:26]:
```suggestion
            }
```

File: src/code/code_800C3C20.c [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+u8 D_8012D200[] = {
+    0, 1, 2, 3, 4, 5, 6,
+};
+
+void func_800C3C20() {
+    u8* iter;
+    for (iter = D_8012D200; iter < D_8012D200 + 7; iter++) {
+        func_800F87A0(*iter);
+    }
```
Comment (Roman971) [inline src/code/code_800C3C20.c:11]: Could the loop be written like this?
```suggestion
    for (i = 0; i < 7; i++) {
        func_800F87A0(D_8012D200[i]);
    }
```

File: src/code/code_800C3C20.c [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#include <global.h>
+
+u8 D_8012D200[] = {
+    0, 1, 2, 3, 4, 5, 6,
+};
+
+void func_800C3C20() {
```
Comment (Roman971) [inline src/code/code_800C3C20.c:7]:
```suggestion
void func_800C3C20(void) {
```

File: include/functions.h [hunk 4]
```diff
@@ -1566,7 +1566,7 @@ u32 THA_IsCrash(TwoHeadArena* tha);
 void THA_Init(TwoHeadArena* tha);
 void THA_Ct(TwoHeadArena* tha, void* ptr, u32 size);
 void THA_Dt(TwoHeadArena* tha);
-// ? func_800C3C20(?);
+void func_800C3C20();
```
Comment (Roman971) [inline include/functions.h:1569]:
```suggestion
void func_800C3C20(void);
```

File: src/code/padsetup.c (hunk 1 above)
Comment (Random06457) [reply src/code/padsetup.c:26]: The empty else is required to match.

File: src/code/code_800C3C20.c (hunk 2 above)
Comment (Random06457) [reply src/code/code_800C3C20.c:11]: This didn't match for some reason. (There was an issue with the compare if I recall correctly).

File: src/code/padsetup.c (hunk 1 above)
Comment (Roman971) [reply src/code/padsetup.c:26]: Required how? What does it change in codegen?

Comment (Random06457) [reply src/code/padsetup.c:26]: The empty `else` adds a branch instruction.

File: src/code/z_jpeg.c [hunk 5]
```diff
@@ -97,7 +97,7 @@ u16 Jpeg_GetU16(u8* ptr) {
     if (((u32)ptr & 1) == 0) { // if the address is aligned to 2
         return *(u16*)ptr;
     } else {
-        return *(u16*)(ptr - 1) << 8 | (*(u16*)(ptr + 1) >> 8); // ?? it's exactly like *(16*)ptr
+        return *(u16*)(ptr - 1) << 8 | (*(u16*)(ptr + 1) >> 8); // lbu crashes with unaligned addresses
```
Comment (Roman971) [inline src/code/z_jpeg.c:100]:
```suggestion
        return *(u16*)(ptr - 1) << 8 | (*(u16*)(ptr + 1) >> 8); // lhu crashes with unaligned addresses
```

File: src/code/code_800C3C20.c (hunk 2 above)
Comment (Roman971) [reply src/code/code_800C3C20.c:11]: There must be way to write it without using the address of D_8012D200...
What about if it's a single line?

File: src/code/padsetup.c (hunk 1 above)
Comment (Roman971) [reply src/code/padsetup.c:26]: It's most likely a fake match but I guess there is no way to know for sure. Might be worth trying a switch on errno or something slightly different tho, like an early continue.

Comment (Random06457) [reply src/code/padsetup.c]: Using a switch worked.

**Pull request merged**